### PR TITLE
MAT-8010: Convert Editor to npm package

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-8010](https://jira.cms.gov/browse/MAT-8010)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: local-dev-env/hosted-importmap.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "imports": {
     "@madie/root-config": "//localhost:9000/madie-root-config.js",
     "@madie/madie-layout": "//localhost:8500/madie-madie-layout.js",
```
Comment (RohitKandimalla) [inline local-dev-env/hosted-importmap.json]: Since this is not a singleSPA now. Do we even need this importmap.json ?

File: src/__mocks__/mockServiceConfig.ts [hunk 2]
```diff
@@ -13,4 +15,10 @@ export const mockServiceConfig: ServiceConfig = {
   cqlLibraryService: {
     baseUrl: "library-service.com",
   },
+  fhirService: {
```
Comment (RohitKandimalla) [inline src/__mocks__/mockServiceConfig.ts:18]: I don't think we should ever be hitting fhir-service directly ?

File: src/CqlBuilderPanel/codesSection/useCodeSystems.ts [hunk 3]
```diff
@@ -1,18 +1,21 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useRef } from "react";
 import useTerminologyServiceApi, {
   CodeSystem,
 } from "../../api/useTerminologyServiceApi";
 
 export function useCodeSystems() {
+  console.log("callingUseCodeSystem");
```
Comment (RohitKandimalla) [inline src/CqlBuilderPanel/codesSection/useCodeSystems.ts]: debugging statement ?

File: src/CqlBuilderPanel/codesSection/useCodeSystems.ts [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
   const [codeSystems, setCodeSystems] = useState<CodeSystem[]>([]);
   // const [loading, setLoading] = useState(true); maybe add later since this call is slow
 
+  const terminologyService = useRef(useTerminologyServiceApi());
+
   useEffect(() => {
     const fetchCodeSystems = async () => {
       // eslint-disable-next-line
-      const terminologyService = await useTerminologyServiceApi();
       // eslint-disable-next-line
```
Comment (RohitKandimalla) [inline src/CqlBuilderPanel/codesSection/useCodeSystems.ts]: Can we remove this eslint disable ?

File: src/CqlBuilderPanel/ValueSets/ValueSets.tsx [hunk 5]
```diff
@@ -40,8 +41,7 @@ export default function ValueSets(props: ValueSetsProps) {
       }
     });
     // eslint-disable-next-line
```
Comment (RohitKandimalla) [inline src/CqlBuilderPanel/ValueSets/ValueSets.tsx]: same here

File: webpack.config.js [hunk 6]
```diff
@@ -1,93 +1,73 @@
-const { mergeWithRules } = require("webpack-merge");
-const singleSpaDefaults = require("webpack-config-single-spa-react-ts");
-const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");
+const path = require("path");
 
-const orgName = "madie";
+module.exports = (env, argv) => ({
```
Comment (RohitKandimalla) [inline webpack.config.js:3]: Do we even need Webpack ?

Comment (mcmcphillips) [reply webpack.config.js:3]: we're bundling it with webpack

File: src/CqlBuilderPanel/ValueSets/ValueSets.tsx (hunk 5 above)
Comment (mcmcphillips) [reply src/CqlBuilderPanel/ValueSets/ValueSets.tsx]: removed

**Review by RohitKandimalla — DISMISSED** [review]

File: local-dev-env/hosted-importmap.json (hunk 1 above)
Comment (mcmcphillips) [reply local-dev-env/hosted-importmap.json]: removed

File: src/CqlBuilderPanel/codesSection/useCodeSystems.ts (hunk 3 above)
Comment (mcmcphillips) [reply src/CqlBuilderPanel/codesSection/useCodeSystems.ts]: removed

**Review by RohitKandimalla — APPROVED** [review]
